rules/pccrule.go

```diff
@@ -0,0 +1,145 @@
+//Package rules ...
+// Copyright (c) 2017 Intel Corporation.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+package rules
+
+import (
+	"github.com/intel-go/nff-go/common"
+	"github.com/intel-go/nff-go/flow"
+	"gopkg.in/ini.v1"
+	"strconv"
+)
+
+//PCCFilter ...
+type PCCFilter struct {
+	PCCFilterIdx       int
+	RuleName           string
+	RatingGroup        int
+	ServiceID          string //int
+	RuleStatus         int
+	GateStatus         int //1-Accept, 0-Drop
+	SessionCont        int
+	ReportLevel        int
+	ChargingMode       int
+	MeteringMethod     int
+	MuteNotify         int
+	MonitoringKey      int
+	SponsorID          string //int
+	RedirectInfo       int
+	Precedence         int //(0-255, 255 is lowest)
+	DropPktCount       int
+	UlMbrMtrProfileIdx int //- Specify the meter profile index from meter_profile.cfg
+	DlMbrMtrProfileIdx int
+	SDFFilterIdx       []int
+	ADCFilterIdx       int
+}
+
+//DefaultPccFilter ... default pcc rule all all traffic
+var DefaultPccFilter = PCCFilter{
+	PCCFilterIdx: 0,
+	Precedence:   255,
+	GateStatus:   1,
+}
+
+//ParsePCCRules ...
+func ParsePCCRules(filepath string) []PCCFilter {
+	common.LogInfo(common.Info, "Loading PCC Rules...")
+	var pccFilters []PCCFilter
+	cfg, err := ini.Load(filepath)
+	flow.CheckFatal(err)
+
+	noOfFilters, err := cfg.Section(LblGlobal).Key(LblNumPccFilters).Int()
+	flow.CheckFatal(err)
+
+	for i := 1; i <= noOfFilters; i++ {
+		strVal := strconv.Itoa(i)
+
+		section := cfg.Section(LblPccFilter + "_" + strVal)
+		pccFilter := PCCFilter{PCCFilterIdx: i}
+
+		if section.HasKey(LblUlAmbrMtrProfileIdx) {
+			pccFilter.UlMbrMtrProfileIdx, err = section.Key(LblUlAmbrMtrProfileIdx).Int()
+			flow.CheckFatal(err)
```

Comment:
You can CheckFatal only one time after all conditions

Reply:
hmm let me rework it